core/base/inc/TNamed.h

```diff
@@ -48,9 +47,9 @@ class TNamed : public TObject {
             const char  *GetTitle() const override { return fTitle; }
             ULong_t  Hash() const override { return fName.Hash(); }
             Bool_t   IsSortable() const override { return kTRUE; }
-   virtual  void     SetName(const char *name); // *MENU*
-   virtual  void     SetNameTitle(const char *name, const char *title);
-   virtual  void     SetTitle(const char *title=""); // *MENU*
+   virtual  void     SetName(const std::string_view name); // *MENU*
```

Comment:
Those 3 functions are virtual functions that are overload in some places. (you already found many within `ROOT` itself).  It is known to also be overload in user code that have classes deriving (directly or indirectly) from `TNamed`.   This means that this change is *not* backward compatible for the user.  i.e. they need to change their code is order to use a version of ROOT containing this code.  Worse, **__If__** the user has not upgrade their code to use the `override` keyword (very very likely to be true in many cases), the code will still compile but start doing the "wrong" thing: in case where `SetName` is being called for their objects through the `TNamed` virtual interface, it will **no longer** call their override and instead revert to use the default.

A pattern would need to be devised to make sure that even if they don't change their code, their override is still be called.  One such solution (which will take some calendar time to complete and has significant drawbacks) is to (and this may and may not encounter some technical difficulties):

1. For some version of ROOT, let's call it N (where it would be 6.36 if we did this today)
	a.  Add only non virtual overload that takes a `std::string_view` then those the slow thing and create a `std::string` and pass the `c_str()` to the existing interface.
2. For version N+1 make the existing interface deprecated but for this to not cause problem, one also need to:
        b. Make sure that the `std::string_view` overload is prefered to the `const char*` overload even if the parameter is a `const char*` ... maybe (probably not) it can be done by making the interface `protected`.
3. For version N+2 make the `const char*` private so that we get (maybe/hopefully) an error message when user still overloads the `const char*` overload.

And (of course :)), we would have to test all those scenarii before even starting to merge the code.

Hopefully someone as a better idea to ease the transition (where the most important point is to not have a case where the user code silently does the wrong thing).

Reply:
Does the problem would only appeared if they have set the override keyword in their `SetName` ? 

In the case they did something like : 
```c++
class MyClass : public TNamed
{
public:
 void SetName(const char* name) override { /**...**/}
}
```

this code would not compile anymore as there is not void SetName(const char* name) in `TNamed`. While not backward compatible it does not create "wrong" behaviour. 

Does the compiler will not use their own function if they don't use the `override` keyword as before. 

If they did : 
```cpp
class Derived : public TNamed
{
public:
 void SetName(const char* name ){}
}

TNamed* t= new Derived();
t->SetName("name");
```
does actual behaviour would not be to call the TNamed::SetName() already anyway ? So to do the "wrong way" anyway as I imagine user would expect to call the Child function and not the on in Mother in this case.

Reply:
> While not backward compatible it does not create "wrong" behaviour.

That is correct. Still the sudden compilation failures would only be acceptable if there is a very very clear advantage for the change ... in the eye of the users.

> Does the compiler will not use their own function if they don't use the override keyword as before.

The `override` keyword is optional and the current code (as in your 2nd example) will work perfectly (i.e. call `Derived::SetName`).

> does actual behaviour would not be to call the TNamed::SetName() already anyway ?

Nope. See previous answer.

Reply:
To my understanding the second example with call the TNamed one not the Child one if the override keyword is not here.  As the pointer is of TNamed class.

Reply:
I do not understand what you mean.  The execution seen in the example using CompilerExplorer: https://godbolt.org/z/4KE97nrfh shows that the Derived's class `SetName` is being called.  I.e. the `override` keywords is optional, the important part is that the base class function has the `virtual` keyword and that the derived function has the same signature.

Reply:
Yes, Sorry my bad :( I run the wrong test and was confused by the result